diff --git a/src/vobs/VirtualObject.cc b/src/vobs/VirtualObject.cc
index 6c23fd2..a8ad4da 100644
--- a/src/vobs/VirtualObject.cc
+++ b/src/vobs/VirtualObject.cc
@@ -106,7 +106,7 @@ namespace zenkit {
 			this->physics_enabled = static_cast<bool>((bit1 & 0b000000001000000u) >> 6u);
 
 			if (version == GameVersion::GOTHIC_2) {
-				this->anim_mode = static_cast<AnimationType>(bit1 & 0b000000110000000u >> 7u);
+				this->anim_mode = static_cast<AnimationType>((bit1 & 0b000000110000000u) >> 7u);
 				this->bias = static_cast<std::int32_t>((bit1 & 0b011111000000000u) >> 9u);
 				this->ambient = static_cast<bool>((bit1 & 0b100000000000000u) >> 14u);
 
@@ -239,8 +239,8 @@ namespace zenkit {
 			}
 
 			if (version == GameVersion::GOTHIC_2) {
-				bit1 |= (static_cast<uint8_t>(this->anim_mode) & 2) << 7u;
-				bit1 |= (static_cast<uint8_t>(this->bias) & 0b11111) << 13u;
+				bit1 |= (static_cast<uint8_t>(this->anim_mode) & 3) << 7u;
+				bit1 |= (static_cast<uint8_t>(this->bias) & 0b11111) << 9u;
 				bit1 |= this->ambient << 14u;
 
 				pw->write_ushort(bit1);
